File: v1/src/main/java/com/api/v1/people/domain/PersonCrudRepository.java
package com.api.v1.people.domain;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

import java.util.Optional;

public interface PersonCrudRepository extends MongoRepository<Person, String> {

    @Query("{ $or: [ { 'sin': ?0 }, { 'email': ?1 } ] }")
    Optional<Person> findBySinOrEmail(String sin, String email);
}

File: v2/src/main/kotlin/com/api/v2/people/domain/PersonCrudRepository.kt
package com.api.v2.people.domain

import com.api.v2.people.Person
import org.springframework.data.mongodb.repository.Query
import org.springframework.data.mongodb.repository.MongoRepository

interface PersonCrudRepository: MongoRepository<Person, String> {

    @Query("{ '\$or': [ { 'sin': ?0, 'email': ?1 } ] }")
    fun findBySINOrEmail(sin: String, email: String): Person?
}